src/nvim/diff.c

```diff
@@ -1101,20 +1111,104 @@ static int diff_file_internal(diffio_T *diffio)
   return OK;
 }
 
+///
+/// Post-process the user-produced chunks so they fit into diffio_T logic.
+/// Responsible for freeing the chunks list.
+///
+static int translate_chunks(diffio_T *const diffio, list_T *const chunks)
+{
+  // If a custom expression has been used,
+  // translate received user chunks into diffio data.
+  listitem_T *c = tv_list_first(chunks);
+  list_T *i;  // chunk (i)nformation is made of 4 (n)umbers.
+  listitem_T *n;
+  typval_T *c_val;
+  typval_T *n_val;
+  int i_size;
+  long start_a;
+  long count_a;
+  long start_b;
+  long count_b;
+  int success = FAIL;
+  if (c != NULL) {
```

Comment:
```suggestion
  if (c == NULL) {
    return FAIL;
  }
```

Reply:
Well, [IIUC](https://github.com/iago-lito/neovim/blob/70d78aef1516022c5d4b101508fa59e6bdb16b40/src/nvim/eval.c#L1184) this would leak `chunks`, right? But I get the idea: I'll just `goto theend;` instead of `return FAIL;`.

Reply:
Ah yes, didn't see the unref.